boilerplate/app/components/Toggle.tsx

```diff
@@ -12,20 +12,20 @@ import {
   View,
   ViewStyle,
 } from "react-native"
-import Animated, { FadeIn, useAnimatedStyle, withTiming } from "react-native-reanimated"
+import Animated, { useAnimatedStyle, withTiming } from "react-native-reanimated"
 import { colors, spacing } from "../theme"
-import { iconRegistry } from "./Icon"
+import { iconRegistry, IconTypes } from "./Icon"
 import { Text, TextProps } from "./Text"
 
 type Variants = "checkbox" | "switch" | "radio"
 
-export interface ToggleProps extends Omit<TouchableOpacityProps, "style"> {
+interface BaseToggleProps extends Omit<TouchableOpacityProps, "style"> {
   /**
    * The variant of the toggle.
    * Options: "checkbox", "switch", "radio"
    * Default: "checkbox"
    */
-  variant?: Variants
+  variant?: unknown
```

Comment:
Can this go away since provided in each of the 3 types making up the discriminating union? Genuinely asking for my own understanding as I was trying to write some dynamic props earlier in the week

Reply:
I have this here for the description mainly. If I add descriptions to the 3 Types, then the documentation for the selected discriminating prop will be used.

Reply:
Oh ok cool thanks